core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/aggregation/ConfiguredAggregateColumn.kt

```diff
@@ -1,27 +1,37 @@
 package org.jetbrains.kotlinx.dataframe.impl.aggregation
 
-import org.jetbrains.kotlinx.dataframe.columns.*
+import org.jetbrains.kotlinx.dataframe.columns.ColumnPath
+import org.jetbrains.kotlinx.dataframe.columns.ColumnResolutionContext
+import org.jetbrains.kotlinx.dataframe.columns.ColumnSet
+import org.jetbrains.kotlinx.dataframe.columns.ColumnWithPath
+import org.jetbrains.kotlinx.dataframe.columns.ColumnsResolver
+import org.jetbrains.kotlinx.dataframe.columns.shortPath
 
 internal class ConfiguredAggregateColumn<C> private constructor(
-    val columns: ColumnSet<C>,
+    val columns: ColumnsResolver<C>,
     private val default: C? = null,
-    private val newPath: ColumnPath? = null
+    private val newPath: ColumnPath? = null,
 ) : ColumnSet<C> {
 
-    private fun ColumnWithPath<C>.toDescriptor(keepName: Boolean) = when (val col = this) {
-        is AggregateColumnDescriptor<C> -> {
-            val path = if (keepName) newPath?.plus(col.newPath ?: col.column.shortPath()) ?: col.newPath
-            else newPath ?: col.newPath
-            AggregateColumnDescriptor(col.column, default ?: col.default, path)
+    private fun ColumnWithPath<C>.toDescriptor(keepName: Boolean): AggregateColumnDescriptor<C> =
+        when (val col = this) {
```

Comment:
I think Tolya wrote it like this to be able to rename `this` to `col`, perform the check and keep the function as an expression. Potentially it could be extended if something else should be done for other inheritors of ColumnWithPath.